apis/network/v1alpha2/vpc_referencers_test.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	"github.com/crossplaneio/crossplane-runtime/pkg/test"
+)
+
+const (
+	mockName      = "mockName"
+	mockNamespace = "mockNamespace"
+	mockVPCID     = "mockVPCID"
+)
+
+var (
+	errBoom = errors.New("boom")
+)
+
+type mockCanReference struct {
+	resource.CanReference
+	ns string
+}
+
+func (c *mockCanReference) GetNamespace() string {
+	return c.ns
+}
+
+type mockReader struct {
+	client.Reader
+	readFn func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error
+}
```

Comment:
No changes needed for the moment, but I think we could update these to use the existing `test.MockClient` implementation, which should satisfy the `client.Reader` interface.